sources/packages/backend/libs/integrations/src/institution-integration/ier12-integration/ier12.processing.service.ts

```diff
@@ -301,12 +318,235 @@ export class IER12ProcessingService {
         totalAssessedCost: assessmentData.totalAssessedCost,
         totalAssessmentNeed: assessmentData.totalAssessmentNeed,
         disbursementSentDate: disbursement.dateSent,
+        applicationEventCode: await this.getApplicationEventCode(
+          pendingAssessment,
+          disbursement,
+        ),
       };
       ier12Records.push(ier12Record);
     }
     return ier12Records;
   }
 
+  /**
+   * Get application event code (i.e current state of an application)
+   * @param assessment assessment of institutions.
+   * @param currentDisbursementSchedule current disbursement schedule.
+   * @returns application event code.
+   */
+  private async getApplicationEventCode(
+    assessment: StudentAssessment,
+    currentDisbursementSchedule: DisbursementSchedule,
+  ): Promise<ApplicationEventCode> {
+    const application = assessment.application;
+    switch (application.applicationStatus) {
+      case ApplicationStatus.Assessment:
+        return this.applicationEventCodeDuringAssessment(
+          application.applicationNumber,
+        );
+
+      case ApplicationStatus.Enrolment:
+        return this.applicationEventCodeDuringEnrolmentAndCompleted(
+          currentDisbursementSchedule.coeStatus,
+        );
+
+      case ApplicationStatus.Completed:
+        return this.applicationEventCodeDuringCompleted(
+          currentDisbursementSchedule,
+          application.student.id,
+        );
+
+      case ApplicationStatus.Cancelled:
+        return ApplicationEventCode.DISC;
+    }
+  }
+
+  /**
+   * Get application event code for an application with assessment status.
+   * @param applicationNumber application number.
+   * @returns application event code.
+   */
+  private async applicationEventCodeDuringAssessment(
+    applicationNumber: string,
+  ): Promise<ApplicationEventCode.REIA | ApplicationEventCode.ASMT> {
+    // Check if the application has more than one submissions.
+    switch (
```

Comment:
I feel that for single conditional operation, using inline if would be more simple. 

```ts
const applicationMultipleSubmission =
      await this.applicationService.hasMultipleApplicationSubmissions(
        applicationNumber,
      );
    return applicationMultipleSubmission
      ? ApplicationEventCode.REIA
      : ApplicationEventCode.ASMT;
```